payment-service/src/main/java/com/modsen/paymentservice/services/PaymentServiceImpl.java

```diff
@@ -0,0 +1,256 @@
+package com.modsen.paymentservice.services;
+
+import com.modsen.paymentservice.dto.*;
+import com.modsen.paymentservice.exceptions.*;
+import com.modsen.paymentservice.models.User;
+import com.modsen.paymentservice.enums.PaymentMethodEnum;
+import com.modsen.paymentservice.repositories.UserRepository;
+import com.modsen.paymentservice.services.interfaces.PaymentService;
+import com.modsen.paymentservice.util.ExceptionMessage;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.model.*;
+import com.stripe.param.CustomerCreateParams;
+import com.stripe.param.CustomerUpdateParams;
+import com.stripe.param.PaymentIntentConfirmParams;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor
+public class PaymentServiceImpl implements PaymentService {
+
+    @Value("${stripe.secret}")
+    private String SECRET_KEY;
+
+    @Value("${stripe.public}")
+    private String PUBLIC_KEY;
+
+    private final UserRepository userRepository;
+
+    public CustomerResponse createCustomer(CustomerRequest customerRequest){
+        checkCustomerAlreadyExist(customerRequest.getPassengerId());
+        Customer customer = createStripeCustomer(customerRequest);
+
+        createPayment(customer.getId());
+        saveUserToDatabase(customerRequest.getPassengerId(), customer.getId());
+
+        return CustomerResponse.builder()
+                .id(customer.getId())
+                .email(customer.getEmail())
+                .phone(customer.getPhone())
+                .name(customer.getName())
+                .build();
+    }
+
+    public TokenDto generateTokenByCard(CardRequest cardRequest) {
+        try {
+            Stripe.apiKey = PUBLIC_KEY;
```

Comment:
Вместо `Stripe.apiKey = PUBLIC_KEY;` лучше используй `RequestOptions requestOptions = RequestOptions.builder()
                    .setApiKey(PUBLIC_KEY)
                    .build();` . В остальных методах тоже
```suggestion
             try {
            RequestOptions requestOptions = RequestOptions.builder()
                    .setApiKey(PUBLIC_KEY)
                    .build();

            Map<String, Object> cardParams = createCardParams(cardRequest);
            Token token = createToken(cardParams, requestOptions);
            return new TokenDto(token.getId());
        } catch (StripeException ex) {
            throw new TokenException(ExceptionMessage.GENERATION_TOKEN_EXCEPTION + ex.getMessage());
        }
```